flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -1174,6 +1161,16 @@ public void configure(ReadableConfig configuration, ClassLoader classLoader) {
                                                 == JobManagerOptions.SchedulerType.AdaptiveBatch));
     }
 
+    @Internal
+    public <T> T get(ConfigOption<T> option) {
+        return configuration.get(option);
+    }
+
+    @Internal
+    public <T> Optional<T> getOptional(ConfigOption<T> option) {
+        return configuration.getOptional(option);
+    }
```

Comment:
hmm. Not quite sold on these yet. I see the long-term benefits of making a switch to a plain `Configuration` easier, but I'm worried that someone will try to use `getOptional` for an option that is never in the ExecutionConfig.

Reply:
Yeah, but the only alternative is to replace them with `boolean ExecutionConfig#isPresent(ConfigOption<T>)`.  Take a look how I need to use those methods in `Task.java` :(

Reply:
Are there more usages of these methods (or do we intend to add more)? Maybe we don't need such a generic variant in the first place.
(maybe add a `isTaskCancellationTimeoutSet` method instead?)

Alternatively, maybe we could modify the ExecutionConfig post deserialization on the TM side to contain the "final" execution config.

Reply:
As per offline discussion I've replaced those with `toConfiguration()` method.